mat64/product_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import (
+	"math/rand"
+
+	"gopkg.in/check.v1"
+)
+
+type dims struct{ r, c int }
+
+var productTests = []struct {
+	n       int
+	factors []dims
+	product dims
+	panics  bool
+}{
+	{
```

Comment:
Could you add some crazy chain like
````
factors: []dims{{1, 1000}{1000, 2}{2,2}}
````
That way if we change the algorithm, we have an obvious sign of something going wrong.